packages/ceramic-core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -0,0 +1,289 @@
+import CID from "cids";
+import fetch from "node-fetch";
+
+import {EventEmitter} from "events";
+import {AnchorProof} from "../../document";
+import AnchorService from "../anchor-service";
+
+import {ethers} from "ethers";
+import { BaseProvider } from 'ethers/providers';
+import {decode} from "typestub-multihashes";
+import AnchorServiceResponse from "../anchor-service-response";
+
+/**
+ * Describes response from the external Ceramic Anchor Service
+ */
+class ResponseDto {
+    status: string;
+    message: string;
+    scheduledFor?: number;
+    anchorRecord?: {
+        cid: CID | string;
+    };
+}
+
+/**
+ * Simple poller implementation
+ */
+class Poller extends EventEmitter {
+    /**
+     * @param {int} timeout how long should we wait after the poll started?
+     */
+    constructor(private timeout = 100) {
+        super();
+    }
+
+    /**
+     * Poll function which uses setTimeout instead of setInterval
+     */
+    poll(): void {
+        setTimeout(() => this.emit("poll"), this.timeout);
+    }
+
+    /**
+     * On poll listener
+     * @param cb - Callback function
+     */
+    async onPoll(cb: any): Promise<void> {
+        this.on("poll", cb);
+    }
+}
+
+/**
+ * CID-docId pair
+ */
+class CidDoc {
+    constructor(public cid: CID, public docId: string) {
+    }
+}
+
+const DEFAULT_POLL_TIME = 5000; // 5 seconds
+const DEFAULT_MAX_POLL_TIME = 7200000; // 2 hours
+
+/**
+ * Ethereum network configuration
+ */
+class EthNetwork {
+    constructor(public _network: string, public _chain: string, public _networkId: number, public _type: string) {
+    }
+}
+
+/**
+ * Maps Ethereum chain IDs to network configuration
+ */
+const ETH_CHAIN_ID_MAPPINGS: Map<string, EthNetwork> = new Map([
```

Comment:
I couldn't use simple object literal because Typescript made it really messy to get the value by the key. I found out about the `Record<K, T>` type introduced in Typescript 2.1. It's really neat :)